nfp/preprocessing/xtb_preprocessor.py

```diff
@@ -0,0 +1,153 @@
+import json
+from typing import Callable, Dict, Hashable, Optional
+
+import networkx as nx
+import numpy as np
+import rdkit.Chem
+
+try:
+    import tensorflow as tf
+except ImportError:
+    tf = None
+
+from nfp.preprocessing import features
+from nfp.preprocessing.mol_preprocessor import MolPreprocessor, SmilesPreprocessor
+
+
+class xTBPreprocessor(MolPreprocessor):
+    def __init__(
+        self,
+        *args,
+        explicit_hs: bool = True,
+        xtb_atom_features: Optional[Callable[[rdkit.Chem.Atom], Hashable]] = None,
+        xtb_bond_features: Optional[Callable[[rdkit.Chem.Bond], Hashable]] = None,
```

Comment:
We currently don't support these being set to anything other than the defaults (and these are the wrong types anyways). Do we want to allow the user to specify arbitrary xtb features? Or bond features besides wbo?

Reply:
Yeah I just realized I didn't change this. I think it would be better is we keep the options such that the user can specify what they want like some other features or metric to determine bonds?

Reply:
I think `xtb_atom_features` is functional, (the type would be `Optional[List[str]]`), since we later use that list to select elements from the json file to put in the `atom_xtb` arrays.

Did we want to include bond distances, perhaps?

Reply:
I was going to change the type but I added that part in last so I forgot to do it when I pushed it. Yeah the 3D coordinates are written to another file which is an xyz. I can add in in too. 

There are a lot of global features that xtb gives out. I will add them in too!